syntax = "proto3";

package pd.data;

import "geometry.proto";


message SocketTransformMap
{
    map<string, Transform> socket_map = 1;
}

message Value
{
    oneof value {
        bool bool_value = 1;
        int32 int32_value = 2;
        int64 int64_value = 3;
        float float_value = 4;
        double double_value = 5;
        string string_value = 6;
    };
}

message ValueMap
{
    map<string, Value> value_map = 1;
}

message TransformMap
{
    map<uint32, Transform> transform_map = 1;

    map<uint32, SocketTransformMap> socket_transform_map = 2;

    map<uint32, ValueMap> value_map = 3;

    map<uint32, Transform> actor_transform_map = 4;
}